### PR TITLE
Yuv to rgb changes

### LINKED ISSUE #682: Feature request: Add support for visualizing YUV image formats
Most cameras in the field output data in YUV formats. It would be nice to add support for visualizing image msgs of YUV formats and not just RGB. This could be implemented by converting YUV to RGB within rviz and then visualizing it.

So if the encoding field in the Image msg is yuv then do the conversion in rviz2 to be able to see the video feed, https://github.com/ros2/common_interfaces/blob/master/sensor_msgs/msg/Image.msg

### PR DESCRIPTION
**Pull request by cturcotte-qnx**
Updating Foxy to include function to convert YUV format images into RGB inside rviz.

Fixes #682

### CODE REVIEW

**Comment by cturcotte-qnx** [comment]
@asobhy-qnx

File: rviz_default_plugins/include/rviz_default_plugins/displays/image/ros_image_texture.hpp [hunk 1]
```diff
@@ -136,6 +136,8 @@ class ROSImageTexture : public ROSImageTextureIface
 
   uint32_t width_;
   uint32_t height_;
+  uint32_t stride_;
+  std::shared_ptr<std::vector<uint8_t>> bufferptr_;
```
Comment (clalancette) [inline rviz_default_plugins/include/rviz_default_plugins/displays/image/ros_image_texture.hpp]: Same comment as https://github.com/ros2/rviz/pull/733#pullrequestreview-712761739

**Comment by ahcorde** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/15034/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9771/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12724/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/15198/)

**Review by ahcorde — APPROVED** [review]

**Comment by ahcorde** [comment]
CI warnings on MacOS and Windows are unrelated. Merging!

**Pull request merged**